### PR TITLE
Add support to 32-byte chainwork to ThinConverter class

### PR DESCRIPTION
**Pull request by nathanieliov**
## Description

Add support to 32-byte chainwork to ThinConverter class

## How Has This Been Tested?

Unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

### CODE REVIEW

File: src/test/java/co/rsk/federate/adapter/ThinConverterTest.java [hunk 1]
```diff
@@ -4,15 +4,58 @@
...
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 class ThinConverterTest {
 
+    private static final BigInteger NEGATIVE_CHAIN_WORK = BigInteger.valueOf(-1);
+    private static final BigInteger EIGHT_BYTES_WORK_V1 = new BigInteger("ffffffffffffffff", 16); // 8 bytes
```
Comment (julia-zack) [inline src/test/java/co/rsk/federate/adapter/ThinConverterTest.java:24]: why eight? i'd just say something like `BELOW_MAX_WORK_V1`. wdyt?

File: src/test/java/co/rsk/federate/adapter/ThinConverterTest.java [hunk 2]
```diff
@@ -4,15 +4,58 @@
...
+    private static final int height = 200;
     private static int nhash = 0;
 
+    private static final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+    org.bitcoinj.core.NetworkParameters bitcoinCoreParams = org.bitcoinj.params.MainNetParams.get();
+    co.rsk.bitcoinj.core.NetworkParameters bitcoinjThinParams = co.rsk.bitcoinj.params.MainNetParams.get();
+
+    private static final ECKey userKey = ECKey.fromPrivate(BigInteger.valueOf(100));
+
+    public static Stream<Arguments> validChainWorkArgsProvider() {
```
Comment (julia-zack) [inline src/test/java/co/rsk/federate/adapter/ThinConverterTest.java:41]:
```suggestion
    public static Stream<Arguments> validChainWorkV2ArgsProvider() {
```

File: src/test/java/co/rsk/federate/adapter/ThinConverterTest.java [hunk 3]
```diff
@@ -25,61 +68,87 @@ private co.rsk.bitcoinj.core.Sha256Hash createThinHash() {
...
-        co.rsk.bitcoinj.core.StoredBlock thinStoredBlock = ThinConverter.toThinInstance(originalStoredBlock, new BridgeRegTestConstants());
+
+        // act
+        co.rsk.bitcoinj.core.StoredBlock thinStoredBlock = ThinConverter.toThinInstance(originalStoredBlock, bridgeConstants);
+
+        // assert
         assertEquals(chainWork, thinStoredBlock.getChainWork());
         assertEquals(height, thinStoredBlock.getHeight());
         assertArrayEquals(originalBlock.bitcoinSerialize(), thinStoredBlock.getHeader().bitcoinSerialize());
+        assertNull(ThinConverter.toThinInstance(null, bridgeConstants));
```
Comment (julia-zack) [inline src/test/java/co/rsk/federate/adapter/ThinConverterTest.java:85]: why is this here? shouldn't this be in another test? You are not even using the param from the method source.

File: src/test/java/co/rsk/federate/adapter/ThinConverterTest.java (hunk 2 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/adapter/ThinConverterTest.java:41]: No. This includes both formats of chainwork.

File: src/test/java/co/rsk/federate/adapter/ThinConverterTest.java (hunk 3 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/adapter/ThinConverterTest.java:85]: You right. It can be on its own test.

File: src/test/java/co/rsk/federate/adapter/ThinConverterTest.java (hunk 1 above)
Comment (nathanieliov) [reply src/test/java/co/rsk/federate/adapter/ThinConverterTest.java]: Done

File: src/test/java/co/rsk/federate/adapter/ThinConverterTest.java (hunk 2 above)
Comment (julia-zack) [reply src/test/java/co/rsk/federate/adapter/ThinConverterTest.java:40]: how is the too large work v1 valid in v1?

**Review by jeremy-then — APPROVED** [review]
LGTM.

**Review by julia-zack — APPROVED** [review]

**Comment by nathanieliov** [comment]
Closing this PR in benefit of #359 using the same branch name as rskj branch
